opal/models.py

```diff
@@ -176,42 +222,44 @@ def get_field_enum(cls, name):
             return [i[1] for i in choices]
 
     @classmethod
+    def build_schema_for_field(cls, fieldname):
+        getter = getattr(cls, 'get_field_type_for_' + fieldname, None)
```

Comment:
++ good idea to split this out, although again there is a case for moving this functionality to `opal.core.schemas` if we're moving it around ?

I think we might like to have a referencedata documentation entry for it and at least some unittests for this individually ?